### ISSUE TITLE
gulp.watch bug - (semi-important)

### ISSUE THREAD

**Issue by ORESoftware** [gulpjs/gulp#1224]
I discovered a bug/feature on a Macbook Pro using the Webstorm IDE

this regards code like so:

```javascript
gulp.task('watch',function(){

    var filepath = './public/static/app/templates';

    function meta(){
        grm(filepath, 'text!app/', '', false, './public/static/app/js/meta/allTemplates.js');
    }

    gulp.watch(filepath+'/**/*.*').on('change',function(file){
        console.log('changes to templates!');
        meta();
    });

    meta();
});
```

I have good luck using gulp.watch, but there seems to be a bug when I rename files in Webstorm.

The problem that I was having that made me investigate this further was that when I renamed a file in Webstorm, the change event pertaining to the watch would fire once, but no subsequent changes to the file would fire, whether it was a rename or an edit inside the file. Whereas if I never renamed the file and made many edits inside the file, it would fire the right number of change events.

So in other words, the bug is that anytime I rename a file with Webstorm, only one change event will ever fire for that file ever again for the same gulp process.

Is this bug a known bug?

**Comment by phated** [comment]
Going to chalk this up as a duplicate of #651 - The file watcher has tons of problems.

**Comment by ORESoftware** [comment]
aight, I took a quick scan of #651 and I didn't see this specific one
